plugins/woocommerce/src/StoreApi/Routes/V1/Agentic/CheckoutSessionsComplete.php

```diff
@@ -0,0 +1,430 @@
+	}
+
+	/**
+	 * Get the path regex for this REST route.
+	 *
+	 * @return string
+	 */
+	public static function get_path_regex() {
+		return '/checkout_sessions/(?P<checkout_session_id>[a-zA-Z0-9._-]+)/complete';
+	}
+
+	/**
+	 * Get method arguments for this REST route.
+	 *
+	 * @return array An array of endpoints.
+	 */
+	public function get_args() {
+		return [
+			'args'   => [
+				'checkout_session_id' => [
+					'description' => __( 'The checkout session ID (Cart-Token JWT).', 'woocommerce' ),
+					'type'        => 'string',
+				],
+			],
+			[
+				'methods'             => \WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'get_response' ],
+				'permission_callback' => [ $this, 'is_authorized' ],
+				'args'                => $this->get_complete_params(),
+			],
+			'schema' => [ $this->schema, 'get_public_item_schema' ],
+		];
+	}
+
+	/**
+	 * Get the parameters for completing a checkout session.
+	 *
+	 * @return array Parameters array.
+	 */
+	protected function get_complete_params() {
+		$shared_params = AgenticCheckoutUtils::get_shared_params();
+
+		return [
+			'buyer'        => $shared_params['buyer'],
+			'payment_data' => [
+				'description' => __( 'Payment data including token and provider.', 'woocommerce' ),
+				'type'        => 'object',
+				'properties'  => [
+					'token'           => [
+						'description' => __( 'Payment token from the payment provider.', 'woocommerce' ),
+						'type'        => 'string',
+					],
+					'provider'        => [
+						'description' => __( 'Payment provider identifier.', 'woocommerce' ),
+						'type'        => 'string',
+						'enum'        => [ 'stripe' ],
+					],
+					'billing_address' => $shared_params['fulfillment_address'],
+				],
+				'required'    => [ 'token', 'provider' ],
+			],
+		];
+	}
+
+	/**
+	 * Check if the request is authorized.
+	 *
+	 * Checks feature enablement and cart token validity.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return bool|\WP_Error True if authorized, WP_Error otherwise.
+	 */
+	public function is_authorized( \WP_REST_Request $request ) {
+		// Check if feature is enabled using helper.
+		$auth_check = AgenticCheckoutUtils::is_authorized( $request );
+		if ( is_wp_error( $auth_check ) ) {
+			return $auth_check;
+		}
+
+		// Additional check for cart token validity.
+		if ( ! $this->has_cart_token( $request ) ) {
+			return new \WP_Error(
+				'woocommerce_rest_invalid_checkout_session',
+				__( 'Invalid or expired checkout session ID.', 'woocommerce' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Use the checkout_session_id as Cart-Token, and set the respective values to HTTP header and request.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return bool|null
+	 */
+	protected function has_cart_token( \WP_REST_Request $request ) {
+		$session_id = $request->get_param( 'checkout_session_id' );
+		if ( is_null( $this->has_cart_token ) ) {
```

Comment:
Nit: should we also check that `$session_id` is not empty before we validate the token?

Reply:
I don't think we need it, as it would be validated in the path declare: 

https://github.com/woocommerce/woocommerce/blob/513f8f3d6cb560726139b7ecc700e21fff500ae2/plugins/woocommerce/src/StoreApi/Routes/V1/Agentic/CheckoutSessionsComplete.php#L93-L95